app/models/thread.php

```diff
@@ -0,0 +1,114 @@
+<?php
+class Thread extends AppModel
+{
+    const MIN_STRING_LENGTH = 1;
+    const MAX_STRING_LENGTH = 80;
+
+    public $validation = array(
+       'title' => array(
+            'length' => array(
+               'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH,
+            ),
+        ),
+    );
+
+    public static function getAll($offset, $limit)
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
```

Comment:
I think what you can do is to assign the `LIMIT` and `OFFSET` clauses in a variable and use `sprintf()` to replace the question marks with the appropriate values. The resulting string will then be appended to the query.

Looking into it, this is the peculiar behavior of the PDO objects treating parameters as strings not into its appropriate data type.